rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -2772,6 +2774,52 @@ def test_upgrade_db_32_to_33(user_data_dir):  # pylint: disable=unused-argument
     assert_tx_hash_is_bytes(old=old_history_events, new=new_history_events, tx_hash_index=1, is_history_event=True)  # noqa: E501
 
 
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_upgrade_db_33_to_34(user_data_dir):  # pylint: disable=unused-argument  # noqa: E501
+    """Test upgrading the DB from version 33 to version 34.
+
+    - Change the combined_trades_view so a valid string is returned in the link field instead
+    of a blob.
+    """
+    msg_aggregator = MessagesAggregator()
+    _use_prepared_db(user_data_dir, 'v33_rotkehlchen.db')
+    db_v33 = _init_db_with_target_version(
+        target_version=33,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    with db_v33.conn.read_ctx() as cursor:
+        trades, filter_total_found = db_v33.get_trades_and_limit_info(
+            cursor=cursor,
+            filter_query=TradesFilterQuery.make(),
+            has_premium=True,
+        )
+
+    assert filter_total_found == 7
+    with pytest.raises(UnicodeDecodeError):
+        for trade in trades:
+            TradeSchema().dump(trade)
+    assert HexBytes(trades[-1].link).hex() == '0xb1fcf4aef6af87a061ca03e92c4eb8039efe600d501ba288a8bae90f78c91db5'  # noqa: E501
+
+    # Execute upgrade
+    db = _init_db_with_target_version(
+        target_version=34,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    with db.conn.read_ctx() as cursor:
+        trades, filter_total_found = db_v33.get_trades_and_limit_info(
```

Comment:
Shouldn't this be the new db? It only works because you are passing the cursor from the new DB ... but calling it on the object of the old DB.
```suggestion
        trades, filter_total_found = db.get_trades_and_limit_info(
```